python/examples/atomistic/3-atomistic-model-with-nl.py

```diff
@@ -0,0 +1,300 @@
+equilibrated liquid argon atoms at a temperature of 94.4 K and a mass density of
+1.374 g/cm³ to obtain the pair-correlation function :math:`g(r)` of
+the liquid.
+"""
+
+# sphinx_gallery_thumbnail_number = 2
+
+from typing import Dict, List, Optional
+
+# tools for analysis and visualization
+import ase.geometry.rdf
+
+# tools to run the simulation
+import ase.io
+import ase.md
+import ase.visualize.plot
+import chemiscope
+import matplotlib.pyplot as plt
+
+# the usual suspects
+import numpy as np
+import torch
+
+from metatensor.torch import Labels, TensorBlock, TensorMap
+from metatensor.torch.atomistic import (
+    MetatensorAtomisticModel,
+    ModelCapabilities,
+    ModelMetadata,
+    ModelOutput,
+    NeighborListOptions,
+    System,
+)
+
+# Integration with ASE calculator for metatensor atomistic models
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+
+
+# %%
+#
+# A Lennard-Jones Model
+# ---------------------
+#
+# The Lennard-Jones potential is a mathematical model to approximate the interaction
+# between a pair of neutral atoms or molecules. It is given by the equation:
+#
+# .. math:: V(r) = 4\epsilon \left[ \left( \frac{\sigma}{r} \right)^{12} - \left(
+#     \frac{\sigma}{r} \right)^{6} \right]
+#
+# where :math:`\epsilon` is the depth of the potential well, :math:`\sigma` is the
+# finite distance at which the inter-particle potential is zero, and :math:`r` is the
+# distance between the particles. The 12-6 form is chosen because the :math:`r^12` term
+# approximates the Pauli repulsion at short ranges, while the :math:`r^6` term
+# represents the attractive van der Waals forces. The :math:`r^{12}` is choosen becuase
+# it is just the square of the :math:`r^6` and therefore allows fast evaluation.
+#
+# A Lennard-Jones potential is well-suited for argon because it accurately represents
+# the van der Waals forces that dominate the interactions between argon atoms especially
+# as for all noble gases. This potential was used in one of the first MD simulations:
+# "Correlation in the motions of atoms in liquid Argon" by A. Rahman (`Phys. Rev. 136,
+# A405-A411, 1964 <https://link.aps.org/doi/10.1103/PhysRev.136.A405>`_), which
+# demonstrated the effectiveness of continuous potentials in MD.
+#
+# The model below is a simplified version of a `more complex Lennard-Jones model
+# <https://github.com/Luthaf/metatensor-lj-test/blob/main/src/metatensor_lj_test/pure.py>`_.
+# The linked version also implements ``per_atom`` energies as well as atom selection
+# using the ``selected_atoms`` parameter of the ``forward()`` method. In this model, we
+# shift the energy by its value at the ``cutoff``. This will break the conservativeness
+# of the potential, which is unproblematic in here because we use a large cutoff and
+# therefore the potential already almost decayed to zero. For more sophisticated methods
+# like a polynomial switching potential like XPLOR, we refer to the literature.
+
+
```

Comment:
before showing an implementation of the model, I would try to explain a bit more how the data is stored in neighbors, and how to use it.

Reply:
Good point. I will add a section Metatensor Neighbor lists before this one.